test/shopify_app/controller_concerns/login_protection_test.rb

```diff
@@ -44,9 +44,29 @@ class LoginProtectionControllerTest < ActionController::TestCase
   tests LoginProtectionController
 
   setup do
+    @shop = "my-shop.myshopify.com"
+    @api_key = "api_key"
+    @secret = "secret"
+    @old_secret = "old_secret"
+
     ShopifyApp::SessionRepository.shop_storage = ShopifyApp::InMemoryShopSessionStore
     ShopifyApp::SessionRepository.user_storage = ShopifyApp::InMemoryUserSessionStore
 
+    @session = ShopifyAPI::Auth::Session.new(shop: @shop)
+    ShopifyApp::SessionRepository.store_shop_session(@session)
+
+    ShopifyAPI::Context.setup(
```

Comment:
I believe the setup call is already done for each test by the [test configurer](https://github.com/Shopify/shopify_app/blob/fd32ea008219ad657a444bbed3b7c64d1abe1285/test/dummy/config/initializers/shopify_app.rb#L3) in the [default test setup](https://github.com/Shopify/shopify_app/blob/main/test/test_helper.rb#L48). That way, tests can affect the configuration by tweaking the appropriate settings, like in the [callback controller test](https://github.com/Shopify/shopify_app/blob/main/test/controllers/callback_controller_test.rb#L27).

You should be able to use those defaults from `ShopifyApp.configuration` to create the JWT here!